### PR TITLE
Add utility methods to update csp-config

### PR DESCRIPTION
**Pull request by mbelur**

### CODE REVIEW

File: tests/billing_test_utils.go [hunk 1]
```diff
@@ -27,6 +30,12 @@ const (
 	CspBillingAdapterNameSpace       = "cattle-csp-billing-adapter-system"
 )
 
+var ConfigMapGroupVersionResource = schema.GroupVersionResource{
```
Comment (igomez06) [inline tests/billing_test_utils.go:33]: you can import this from the framework,

File: tests/billing_test_utils.go [hunk 2]
```diff
@@ -196,3 +205,90 @@ func getCSPAdapterCacheManagedNodeCount(client *rancher.Client) (int, error) {
...
+	// Convert the UnstructuredConfigMap to TypedConfigMap
+	var typedConfigMap coreV1.ConfigMap
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(configMapUnStructured.Object, &typedConfigMap)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// Unmarshal
+	jsonData := map[string]interface{}{}
+	err = json.Unmarshal([]byte(typedConfigMap.Data["data"]), &jsonData)
```
Comment (igomez06) [inline tests/billing_test_utils.go:235]: Why are you unmarshaling? Isn't Data a `map[string]string`

File: tests/billing_test_utils.go [hunk 3]
```diff
@@ -196,3 +205,90 @@ func getCSPAdapterCacheManagedNodeCount(client *rancher.Client) (int, error) {
...
+
+	if err != nil {
+		return err
+	}
+	return nil
+
+}
+
+// UpdateConfigMap is a helper function that uses the dynamic client to update a config map on a namespace for a specific cluster.
+// This can be moved the framework
```
Comment (igomez06) [inline tests/billing_test_utils.go:261]: agreed.

**Review by igomez06 — COMMENTED** [review]
few questions

File: tests/billing_test_utils.go (hunk 1 above)
Comment (mbelur) [reply tests/billing_test_utils.go]: Updated

File: tests/billing_test_utils.go (hunk 2 above)
Comment (mbelur) [reply tests/billing_test_utils.go:230]: > Why are you unmarshaling? Isn't Data a `map[string]string`

the typedConfigMap.Data is a map[string]string .. the  typedConfigMap.Data["data"] is the string representation. Let me if there is better way to handle this

Comment (igomez06) [reply tests/billing_test_utils.go:230]: I don't quite understand the intention here ... couldn't you just do 
typedConfigMap.Data["last_billed"] = ago.Format(time.RFC3339)

Comment (mbelur) [reply tests/billing_test_utils.go:230]: > I don't quite understand the intention here ... couldn't you just do typedConfigMap.Data["last_billed"] = ago.Format(time.RFC3339)

This seems to create a new key (last_billed) outside of the data ---> data, last_billed where as the last_billed entry should be part of the data.

**Review by igomez06 — APPROVED** [review]
LGTM

**Pull request merged**